### PR TITLE
Feat (MM): State Change Event 

### LINKED ISSUE #238: Notify on Metadata Update
A mechanism to sent events to the EventBus when metadata has been is created/updated

### PR DESCRIPTION
**Pull request by williamputraintan**
Update Metadata Event State Change for library create/update

Resolves #238 

~Will wait after #566~

### CODE REVIEW

**Comment by reisingerf** [comment]
Nice!

How did you generate the MetadataStateChange code (in `lib/workload/stateless/stacks/metadata-manager/proc/aws/event/schema/orcabus_metadatamanager/metadatastatechange/`)?
That does not look like from EventBridge...

File: docs/schemas/events/metadatamanager/MetadataStateChange.schema.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        "action",
+        "data",
+        "refId"
+      ],
+      "properties": {
+        "model": {
+          "type": "string",
+          "enum": [
+            "LIBRARY",
+            "SPECIMEN",
```
Comment (reisingerf) [inline docs/schemas/events/metadatamanager/MetadataStateChange.schema.json:59]: Until we have a better handle on how specimen work, should this better be `SAMPLE`?

**Review by reisingerf — APPROVED** [review]

File: docs/schemas/events/metadatamanager/MetadataStateChange.schema.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "$id": "https://raw.githubusercontent.com/umccr/orcabus/main/docs/schemas/events/workflowmanager/WorkflowRunStateChange.schema.json",
```
Comment (victorskl) [inline docs/schemas/events/metadatamanager/MetadataStateChange.schema.json:3]: Need change schema ID.

```
https://raw.githubusercontent.com/umccr/orcabus/main/docs/schemas/events/metadatamanager/MetadataStateChange.schema.json
```

File: docs/schemas/events/metadatamanager/example/MSC__example1.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  "account": "123456789012",
+  "time": "2024-05-01T09:25:44Z",
+  "region": "ap-southeast-2",
+  "resources": [],
+  "detail": {
+    "model": "LIBRARY",
+    "action": "CREATE",
+    "refId": "lib.01FZ7N3XK1GZ5V9P8Z4H7J5S9F",
+    "data": {
+      "orcabus_id": "lib.01FZ7N3XK1GZ5V9P8Z4H7J5S9F",
```
Comment (victorskl) [inline docs/schemas/events/metadatamanager/example/MSC__example1.json:15]: Would you be able to support `camelCase` for event alignment? Convention is the same as API, once broadcast out from the app...

**Review by victorskl — APPROVED** [review]
Looks good all in all. Few comments to attend.

File: docs/schemas/events/metadatamanager/MetadataStateChange.schema.json (hunk 1 above)
Comment (williamputraintan) [reply docs/schemas/events/metadatamanager/MetadataStateChange.schema.json:59]: Ahh yes, this is probably from the draft before we changed it to `SAMPLE`, I suppose we could just put `LIBRARY` for now as this is the only event we publish.

File: docs/schemas/events/metadatamanager/example/MSC__example1.json (hunk 3 above)
Comment (williamputraintan) [reply docs/schemas/events/metadatamanager/example/MSC__example1.json:15]: Yes, will look into this. thanks!

File: docs/schemas/events/metadatamanager/MetadataStateChange.schema.json (hunk 1 above)
Comment (reisingerf) [reply docs/schemas/events/metadatamanager/MetadataStateChange.schema.json:59]: Yup, that works too.

**Comment by williamputraintan** [comment]
> How did you generate the MetadataStateChange code (in lib/workload/stateless/stacks/metadata-manager/proc/aws/event/schema/orcabus_metadatamanager/metadatastatechange/)?
That does not look like from EventBridge...

It is from EventBridge, perhaps I just put it at a different folder structure.

**Pull request merged**